src/pages/Navigation/useNavigation.js

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { searchUserRequest } from '../../actions/navigation-actions';
+import useSearchInput from '../../hooks/useSearchInput';
+
+const useNavigation = () => {
+  const dispatch = useDispatch();
+  const searchUsers = useSelector((state) => state.navigation.searchUsers);
+  const isSearching = useSelector((state) => state.navigation.isSearching);
+  const chats = useSelector((state) => state.main.chats);
+  const isLoadedChats = useSelector((state) => state.main.isLoaded);
+  const [filteredUsers, setFilteredUsers] = useState(searchUsers);
+  const [suggested, setSuggested] = useState([]);
+
+  const {
+    handleSearchDebounce,
+    handleSearchSubmit,
+    clearSearchValue,
+    handleSearchChange,
+    isTyping,
+    searchValue,
+  } = useSearchInput((value) => {
+    dispatch(searchUserRequest(value));
+  });
+
+  useEffect(() => {
+    if (searchUsers.length !== 0 && searchValue.length !== 0) {
+      const searchValueLowerCase = searchValue.toLowerCase();
+      const filteredUsersBySearchValue = searchUsers.filter((user) =>
+        user.username.toLowerCase().includes(searchValueLowerCase),
+      );
+      setFilteredUsers(filteredUsersBySearchValue);
+    }
+  }, [searchUsers, searchValue]);
+
+  useEffect(() => {
```

Comment:
you have two use effects that are recalculating based on same searchValue prop. Check https://blog.battlefy.com/how-to-escape-react-hooks-hell-a66c0d142c9e